pipeline-maven/src/main/java/org/jenkinsci/plugins/pipeline/maven/util/MavenVersion.java

```diff
@@ -0,0 +1,64 @@
+package org.jenkinsci.plugins.pipeline.maven.util;
+
+import java.util.Objects;
+
+public class MavenVersion {
```

Comment:
This is fine. `hudson.util.VersionNumber` should also work too if you don't want to maintain a custom class.

Reply:
https://issues.jenkins.io/browse/JENKINS-72516

[withMaven] Options: []
[withMaven] Available options:[withMaven] Options: []
[withMaven] Available options: 
[withMaven] using JDK installation provided by the build agent
[withMaven] using Maven installation 'Maven 3.9.6'
[withMaven] failed to read Maven version: Unable to serialize Optional[Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)]
[withMaven] WARNING: You are running an old version of Maven (UNKNOWN), you should update to at least 3.8.x
 
[withMaven] using JDK installation provided by the build agent
[withMaven] using Maven installation 'Maven 3.9.6'
[withMaven] failed to read Maven version: Unable to serialize Optional[Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)]
[withMaven] WARNING: You are running an old version of Maven (UNKNOWN), you should update to at least 3.8.x